frontend/public/extend/devconsole/utils/pipeline-actions.tsx

```diff
@@ -0,0 +1,207 @@
+}
+
+interface Action {
+  label: string | Object;
+  callback: () => void;
+}
+
+type ActionFunction = (kind: K8sKind, obj: K8sResourceKind) => Action;
+
+const redirectToResourceList = (resource: string) => {
+  const url = location.pathname;
+  let basePath = '/k8s/';
+  if (getPerspective(url) === 'dev') {
+    basePath = '/dev/k8s';
+  }
+  const activeNamespace = getNamespace(url);
+  const resourceUrl =
+    activeNamespace === ALL_NAMESPACES_KEY
+      ? `${basePath}/all-namespaces/${resource}`
+      : `${basePath}/ns/${activeNamespace}/${resource}`;
+  history.push(resourceUrl);
+};
+
+export const newPipelineRun = (pipeline: Pipeline, latestRun: PipelineRun): PipelineRun => {
+  if (
+    (!pipeline || !pipeline.metadata || !pipeline.metadata.name || !pipeline.metadata.namespace) &&
+    (!latestRun ||
+      !latestRun.metadata ||
+      !latestRun.spec ||
+      !latestRun.spec.pipelineRef ||
+      !latestRun.spec.pipelineRef.name)
+  ) {
+    // eslint-disable-next-line no-console
+    console.error(
+      'Unable to create new PipelineRun. Missing "metadata" in ',
+      pipeline,
+      ' and spec.pipelineRef in ',
+      latestRun,
+    );
+    return null;
+  }
+  return {
+    apiVersion: `${PipelineRunModel.apiGroup}/${PipelineRunModel.apiVersion}`,
+    kind: PipelineRunModel.kind,
+    metadata: {
+      name:
+        latestRun && latestRun.metadata && latestRun.metadata.name
+          ? isNaN(parseInt(latestRun.metadata.name[latestRun.metadata.name.length - 1], 10))
+            ? `${latestRun.metadata.name}-1`
+            : [
+              latestRun.metadata.name.slice(0, -1),
+              parseInt(latestRun.metadata.name[latestRun.metadata.name.length - 1], 10) + 1,
+            ].join('')
+          : `${pipeline.metadata.name}-run-1`,
+      namespace:
+        latestRun && latestRun.metadata && latestRun.metadata.namespace
+          ? latestRun.metadata.namespace
+          : pipeline.metadata.namespace || '',
+      labels:
+        latestRun && latestRun.metadata && latestRun.metadata.labels
+          ? latestRun.metadata.labels
+          : {
+            'tekton.dev/pipeline': pipeline.metadata.name,
+          },
+    },
+    spec: {
+      pipelineRef: {
+        name:
+          latestRun &&
+          latestRun.spec &&
+          latestRun.spec.pipelineRef &&
+          latestRun.spec.pipelineRef.name
+            ? latestRun.spec.pipelineRef.name
+            : pipeline && pipeline.metadata && pipeline.metadata.name
+              ? pipeline.metadata.name
+              : null,
+      },
+      params:
+        latestRun && latestRun.spec && latestRun.spec.params
+          ? latestRun.spec.params
+          : pipeline.spec && pipeline.spec.params
+            ? pipeline.spec.params
+            : null,
+      trigger: {
+        type: 'manual',
+      },
+    },
+  };
+};
+
+export const triggerPipeline = (
+  pipeline: Pipeline,
+  latestRun: PipelineRun,
+  redirectURL: string,
```

Comment:
```suggestion
  redirectURL?: string,
```
So you can provide a null value instead of an empty string.